### PR TITLE
tests: make valkey_search integration tests optional

### PR DESCRIPTION
**Pull request by vyavdoshenko**
Make valkey_search module load gracefully when integration/ directory is not synced.
This allows pytest to run without errors when valkey-search tests haven't been synced via `sync-valkey-search-tests.sh`.

Related to: https://github.com/dragonflydb/dragonfly/pull/6320

### CODE REVIEW

File: tests/dragonfly/valkey_search/conftest.py [hunk 1]
```diff
@@ -2,9 +2,21 @@
...
 from .. import dfly_args
 
+# Check if integration directory exists
+_integration_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "integration")
+INTEGRATION_AVAILABLE = os.path.isdir(_integration_dir)
+
+
+def pytest_ignore_collect(collection_path, config):
+    """Skip collection of integration tests if directory is not synced."""
+    if not INTEGRATION_AVAILABLE and "integration" in str(collection_path):
```
[the remark this answers is not in the document]
Comment (vyavdoshenko) [reply tests/dragonfly/valkey_search/conftest.py:16]: The check is scoped to the valkey_search conftest, so it only affects tests collected under this directory. The chance of having another unrelated "integration" directory here is minimal. The suggested relative_to() approach with multiple try/except blocks adds significant complexity for a very unlikely edge case.

File: tests/dragonfly/valkey_search/__init__.py [hunk 2]
```diff
@@ -9,62 +9,70 @@
...
-# Add all modules to sys.modules
-sys.modules["valkeytestframework"] = valkey_test_framework
-sys.modules["valkeytestframework.valkey_test_case"] = valkey_test_case
-sys.modules["valkeytestframework.util"] = util_module
-sys.modules["valkeytestframework.util.waiters"] = waiters_module
-sys.modules["valkeytestframework.conftest"] = conftest_module
+
+# Check if integration directory exists before attempting import
+_integration_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "integration")
+INTEGRATION_AVAILABLE = os.path.isdir(_integration_dir)
```
[the remark this answers is not in the document]
Comment (vyavdoshenko) [reply tests/dragonfly/valkey_search/__init__.py:15]: Cannot import from conftest.py into __init__.py - conftest is a pytest plugin, not a regular module. This would also create a circular dependency since conftest imports from the parent package via from .. import dfly_args. Two lines of simple os.path code don't warrant a separate shared module.

File: tests/dragonfly/valkey_search/__init__.py [hunk 3]
```diff
@@ -9,62 +9,70 @@
...
+else:
+    from .integration import compatibility
+
+    # Add current directory to path for imports
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    if current_dir not in sys.path:
+        sys.path.insert(0, current_dir)
+
+    # Import the Dragonfly-specific test case classes
+    exec(open(os.path.join(current_dir, "valkey_search_test_case_dragonfly.py")).read())
```
[the remark this answers is not in the document]
Comment (vyavdoshenko) [reply tests/dragonfly/valkey_search/__init__.py]: fixed

**Review by romange — APPROVED** [review]
